### ISSUE TITLE
syntax error, unexpected VEC2, expecting COMMA or SEMICOLON when trying to run

### ISSUE THREAD

**Issue by BTCallahan** [armory3d/armory#1532]
**Description**
When I attempt to play the .blend file, I get the following error message:

`ERROR: /home/username/Documents/Blender/armory testing/water move example/build_water movement example/compiled/Shaders/Red_mesh.frag.glsl:17: '' :  syntax error, unexpected VEC2, expecting COMMA or SEMICOLON
ERROR: 1 compilation errors.  No code generated.

SPIR-V is not generated for failed compile or link

Compiling shader 5 of 31 (Red_mesh.frag.glsl) failed:
Shader compiler error.
(node:21306) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Shader compiler error.
(node:21306) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
ERROR: /home/username/Documents/Blender/armory testing/water move example/build_water movement example/compiled/Shaders/Water_mesh.frag.glsl:12: 'None' : undeclared identifier 
ERROR: /home/username/Documents/Blender/armory testing/water move example/build_water movement example/compiled/Shaders/Water_mesh.frag.glsl:12: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.

SPIR-V is not generated for failed compile or link

Compiling shader 10 of 31 (Water_mesh.frag.glsl) failed:
Shader compiler error.
ERROR: /home/username/Documents/Blender/armory testing/water move example/build_water movement example/compiled/Shaders/Water_mesh.vert.glsl:18: 'None' : undeclared identifier 
ERROR: /home/username/Documents/Blender/armory testing/water move example/build_water movement example/compiled/Shaders/Water_mesh.vert.glsl:18: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.

SPIR-V is not generated for failed compile or link

Compiling shader 11 of 31 (Water_mesh.vert.glsl) failed:
Shader compiler error.
Build failed, check console`

When I opened the file Red_mesh.frag.glsl, I found the following text at line 17:

`vec3 texCoordBlend = vec3(0.0); vec2 vec2(Group_TextureCoordinate_texread_Generated_res.x, 1.0 - Group_TextureCoordinate_texread_Generated_res.y)1 = vec2(0.0); vec2 vec2(Group_TextureCoordinate_texread_Generated_res.x, 1.0 - Group_TextureCoordinate_texread_Generated_res.y)2 = vec2(0.0);`

This suggests that the problem is with the part of the program that generates the shader code.

**To Reproduce**
Unzip and play the .blend file.

**Expected behavior**
That the shaders would be compiled without any problems.

**System**
Blender: 2.81a
Armory: 2019.11.0
OS: Fedora 30
Graphics card: GeForce GTX 1650/PCIe/SSE2
[water move example.zip](https://github.com/armory3d/armory/files/4013540/water.move.example.zip)

**Test File**
*(drag & drop the zipped .blend file here)*

**Comment by Sandy10000** [comment]
List of material nodes that can be used in Armory.(https://github.com/armory3d/armory/wiki/supported_nodes)
Unsupported nodes are used for the materials "Red" and "Water". Removing them will solve the problem of "Red_mesh.frag.glsl" and "Water_mesh.frag.glsl".

But then have a problem with "Water_mesh.vert.glsl" but I don't know how to fix this at the moment.
```
ERROR: /compiled/Shaders/Water_mesh.vert.glsl:17: 'None' : undeclared identifier 
ERROR: /compiled/Shaders/Water_mesh.vert.glsl:17: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.
```

**Comment by BTCallahan** [comment]
I apologise for the delay. I thought I had clicked the comment button, but I must have forgotten.

> List of material nodes that can be used in Armory.(https://github.com/armory3d/armory/wiki/supported_nodes)
> Unsupported nodes are used for the materials "Red" and "Water". Removing them will solve the problem of "Red_mesh.frag.glsl" and "Water_mesh.frag.glsl".
> 
> But then have a problem with "Water_mesh.vert.glsl" but I don't know how to fix this at the moment.
> 
> ```
> ERROR: /compiled/Shaders/Water_mesh.vert.glsl:17: 'None' : undeclared identifier 
> ERROR: /compiled/Shaders/Water_mesh.vert.glsl:17: '' : compilation terminated 
> ERROR: 2 compilation errors.  No code generated.
> ```

Ah. Thanks for the info. I've removed that part from the nodes. I do get the same errors as you as well as one that states:
`example/compiled/Shaders/Red_mesh.frag.glsl:17: '' :  syntax error, unexpected VEC2, expecting COMMA or SEMICOLON`
Looking at line 17 of Red_mesh.frag.glsl shows the following:
`vec3 texCoordBlend = vec3(0.0); vec2 vec2(Group_TextureCoordinate_texread_Generated_res.x, 1.0 - Group_TextureCoordinate_texread_Generated_res.y)1 = vec2(0.0); vec2 vec2(Group_TextureCoordinate_texread_Generated_res.x, 1.0 - Group_TextureCoordinate_texread_Generated_res.y)2 = vec2(0.0);`
And at line 27:
`vec3 texCoordBlend = vec3(0.0); vec2 vec2(Geometry_texread_Position_res.x, 1.0 - Geometry_texread_Position_res.y)1 = vec2(0.0); vec2 vec2(Geometry_texread_Position_res.x, 1.0 - Geometry_texread_Position_res.y)2 = vec2(0.0);`
Not only is the complier is putting a 1 or 2 right after the closing bracket, but it's also not defining hte vec2's correctly - it should be something like:
`vec2 thisvec2 = vec2(Geometry_texread_Position_res.x, 1.0 - Geometry_texread_Position_res.y);`

**Comment by Sandy10000** [comment]
I mistakenly considered the "Mix RGB" node as unsupported and removed subsequent nodes. 
I'm sorry to be wrong.

If delete the unsupported node, only "Water_mesh.frag.glsl" problem will be solved.

And about this error
```
example/compiled/Shaders/Red_mesh.frag.glsl:17: '' : syntax error, unexpected VEC2, expecting COMMA or SEMICOLON
```
Was reported here(https://github.com/armory3d/armory/issues/1292). Changing the Projection of the "Image Texture" node from "Box" to "Flat" can avoid the problem.

Avoiding the "Box" problem results in the following error:
```
example/compiled/Shaders/Red_mesh.frag.glsl:51: 'None' : undeclared identifier 
example//compiled/Shaders/Red_mesh.frag.glsl:51: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.
```
This problem can be avoided by replacing the "Vector Math" node with a "Math" node calculation.
![1](https://user-images.githubusercontent.com/48373918/71981251-548f8900-3265-11ea-8443-4b4135d12b18.jpg)
Compiling is possible if you avoid these two problems.

In other words, there seems to be a problem with the "Vector Math" node in addition to the reported # 1292 problem.

**Comment by BTCallahan** [comment]
That fixes most of the errors. It runs now, aside from showing a black screen from time to time. To duplicate the black screen effect, position the mouse cursor in the top half of the screen and wait for a few seconds. In addition, the terminal also outputs the message, `Uniform texUnpack not found.` four times. Unfortunately, it dosen't say where in the shader the error is occurring.
[water move example.zip](https://github.com/armory3d/armory/files/4051937/water.move.example.zip)
